src/modules/discovery/hosts.py

```diff
@@ -87,61 +105,39 @@ def __init__(self, event):
-            for subnet in subnets:
+            for subnet in scan_subnets:
                 if self.event.kubeservicehost and self.event.kubeservicehost in IPNetwork("{}/{}".format(subnet[0], subnet[1])):
                     should_scan_apiserver = False
                 logging.debug("From pod scanning subnet {0}/{1}".format(subnet[0], subnet[1]))
-                for ip in HostDiscoveryHelpers.generate_subnet(ip=subnet[0], sn=subnet[1]):
+                for ip in HostDiscoveryUtils.generate_subnet(ip=subnet[0], sn=subnet[1]):
                     self.publish_event(NewHostEvent(host=ip, cloud=cloud))
             if should_scan_apiserver:
                 self.publish_event(NewHostEvent(host=IPAddress(self.event.kubeservicehost), cloud=cloud))
 
-            
-    def is_azure_pod(self):
-        try:
-            logging.debug("From pod attempting to access Azure Metadata API")
-            if requests.get("http://169.254.169.254/metadata/instance?api-version=2017-08-01", headers={"Metadata":"true"}, timeout=5).status_code == 200:
-                return True
-        except requests.exceptions.ConnectionError:
-            return False
-
-   # for pod scanning
-    def traceroute_discovery(self):
-        external_ip = requests.get("http://canhazip.com").text # getting external ip, to determine if cloud cluster
-        cloud = HostDiscoveryHelpers.get_cloud(external_ip)
-        logging.getLogger("scapy.runtime").setLevel(logging.ERROR) # disables scapy's warnings
-        from scapy.all import ICMP, IP, Ether, srp1
-
-        node_internal_ip = srp1(Ether() / IP(dst="google.com" , ttl=1) / ICMP(), verbose=0)[IP].src
-        return [ [node_internal_ip,"24"], ], external_ip
-
-    # quering azure's interface metadata api | works only from a pod
-    def azure_metadata_discovery(self):
-        logging.debug("From pod attempting to access azure's metadata")
-        machine_metadata = json.loads(requests.get("http://169.254.169.254/metadata/instance?api-version=2017-08-01", headers={"Metadata":"true"}).text)
-        address, subnet= "", ""
-        subnets = list()
-        for interface in machine_metadata["network"]["interface"]:
-            address, subnet = interface["ipv4"]["subnet"][0]["address"], interface["ipv4"]["subnet"][0]["prefix"]
-            logging.debug("From pod discovered subnet {0}/{1}".format(address, subnet if not config.quick else "24"))
-            subnets.append([address,subnet if not config.quick else "24"])
-
-            self.publish_event(AzureMetadataApi(cidr="{}/{}".format(address, subnet)))
-
-        return subnets, "Azure"
+    def pod_subnet_discovery(self):
+        # normal option when running as a pod is to scan it's own subnet
+        # The gateway connects us to the host, and we can discover the 
+        # kubelet from there, other ip's are pods that are running 
+        # next to us,
+        return [(HostDiscoveryUtils.get_default_gateway(), "24")]
```

Comment:
As mentioned above, you can directly return `HostDiscoveryUtils.get_default_gateway(), "24"` instead of a list.